src/seismometer/data/loader/event.py

```diff
@@ -153,7 +160,33 @@ def _merge_event(
         min_leadtime_hrs=offset_hrs,
         window_hrs=window_hrs,
         event_base_time_col="Time",
+        event_base_val_dtype=event_dtype,
         sort=sort,
         merge_strategy=config.events[disp_event].merge_strategy,
         impute_val=impute_val,
     )
+
+
+def _get_source_type(config: ConfigProvider, event: Event) -> str:
+    """Get the dtype of the first source of the specified event."""
+    dtype = None
+    multitypes = set()  # Aggregate woarning messages
+    for source in event.source:
+        source_defn = config.event_defs.get(source, None)
+        new_type = getattr(source_defn, "dtype", None)
```

Comment:
Can we not use getattr here?

Reply:
This one handles source_defn being None from the line above.

Reply:
then shouldn't this be `if source_dfn: new_type = source_defn.dtype`? I guess you save a line, but it feels off.

Reply:
This also needs to have new_type initialized to None (or do the if None else bit for your if)  

Updated to the if block; with walrus since source_defn is only used inside it.